[Login] Fix some issues

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12032 Closes: #12046
<!-- Id number of the GitHub issue this PR addresses. -->

⚠️ Please don't merge until the FluxC hash is updated.

### Description
This PR fixes three issues:
1. #11931 by updating FluxC to include the changes of the FluxC [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3053).
2. #12032 by clearing the cached websites before proceeding with the login flow.
3. #12046 by making sure to show the primary button after finishing with the site fetch.

The three issues have a deeper explanation of the causes of the issues.

### Steps to reproduce
**Use `trunk`**
1. Use a Jetpack connected website.
2. To prepare the website, please install the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin, then add the following snippet:
```PHP
function modify_pre_get_shortlink_defaults($return, $id, $context, $allow_slugs) { 
	if ($id == 0) {
		return home_url( '/' );
	}
    return $return; 
}
// Add the filter to return the `shortlink` type as the `Link` header for the home page as well
add_filter( "pre_get_shortlink", "modify_pre_get_shortlink_defaults", 10, 4 );
```
3. Open the app.
5. Enter the site URL.
6. Use site credentials for signing.
7. Enter some credentials (no matter if they are wrong or right).
8. Notice that the Dialog will show `Fetching website` after a while, **then the login will fail.**
9. Attempt the login using the WebView.
10. **Notice that the login still fails with a generic error.**
11. Go back to the WordPress.com email screen.
12. Sign in using WordPress.com.
13. **Notice that the Account Mismatch screen is shown.**
14. Go back to the Site Picker.
15. **Notice that the Continue button is hidden.**

The three bugs are highlighted in the steps using bold text.
 
### Testing information
#### TC1: Testing the fix of #11931
1. Use the website from the above steps.
2. Test site credentials sign in.
3. Confirm the login works without issues.

#### TC2: Testing the fix of #12032
1. Use a Jetpack connected website (it can be the same website from the above steps)
2. Install the plugin [WPS Hide Login](https://wordpress.org/plugins/wps-hide-login/) on your website.
3. Test site credentials login.
4. When it fails, don't use the WebView login.
5. Go back to the WordPress.com email screen, then proceed with WordPress.com login.
6. Confirm the login works as expected.

#### TC3: Testing the fix of #12046
1. Start the login, and enter the address of a website not connected to your account (tip: you can use `wordpress.com` here 😄)
2. Proceed with WordPress.com login.
3. When the Account Mismatch screen is shown, go back.
4. Confirm the Continue button is shown.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Make sure to clear all cached sites when starting the login flow

hichamboushaba committed:
Restore the primary button after navigating to account mismatch

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/226">19.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Bump fluxc

hichamboushaba committed:
Reset the skeleton visibility before showing sites

This is needed to avoid a race condition where the skeleton is not shown for the Account Mismatch fetch, because we will reset it just after launching the fetch.

hichamboushaba committed:
Add a release notes entry

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt:
```diff
@@ -140,6 +140,8 @@ class SitePickerViewModel @Inject constructor(
         val result = repository.fetchWooCommerceSites()
         val duration = System.currentTimeMillis() - startTime
 
+        sitePickerViewState = sitePickerViewState.copy(isSkeletonViewVisible = false)
```
I just moved this line from below to here, this was to fix an issue where the skeleton wasn't shown when fetching the website information during an account mismatch.
Originally this line was here, I believe moving it below was an accidental change recently.

JorgeMucientes commented:
Excellent job at investing and fixing this multiple issues @hichamboushaba, hope this translates in more users seamlessly logging in. 

One minor thing I noticed when following steps to reproduce the bug from `trunk` I did experience some slight differences in the expected error. The described error was the following: 
> 7. Notice that the Dialog will show Fetching website after a while, then the login will fail.
> 8. Attempt the login using the WebView.
> 9. Notice that the login still fails with a generic error.

Whereas in my case I was redirected to applications passwords accept connection, and then I was stuck there: 

https://github.com/user-attachments/assets/d4c7adc3-42ce-4e52-9e62-249b807b65fa

With the current fix from this PR I didn't experience this anymore, just sharing in case there's some potential log in scenario that we could be missing.

hichamboushaba committed:
Bump fluxc

hichamboushaba commented:
Thanks @JorgeMucientes for the review.

> Whereas in my case I was redirected to applications passwords accept connection, and then I was stuck there:

We are talking about the same thing, I should've probably been clearer, but the `Attempt the login using the WebView` refers to the `applications passwords accept connection` you mentioned.
Now regarding the fact that you couldn't accept the connection, it's because you have an application password with the same name on your site, and there is a tiny inline error that shows in the WebView that says `Each application name should be unique` 😅 (the UI of this web page is totally not user friendly 🤷‍♂️)
